TypeCobol/Compiler/CompilationDocument.cs

```diff
@@ -54,11 +54,23 @@ public class CompilationDocument
         public List<RemarksDirective.TextNameVariation> CopyTextNamesVariations { get; set; }
 
         public List<CopyDirective> MissingCopies { get; set; }
+        /// <summary>
+        /// Is this Compilation Document for a Copy.
+        /// </summary>
+        internal bool IsForCopy
+        {
+            get;
+            set;
```

Comment:
`CompilationUnit` is for programs so this property can be made readonly and set at construction time (default true, false when called from CompilationUnit constructor)